src/api/libcellml/importer.h

```diff
@@ -0,0 +1,177 @@
+
+    /**
+     * @brief Get a model from the library at the given @p index.
+     *
+     * Returns a reference to the model at the index @p index for this
+     * importer's library. If the index is not valid a @c nullptr is returned. The valid
+     * range for the index is [0, \#library items).
+     *
+     * @param index The index of the model to return.
+     *
+     * @return A reference to the model at the given index on success, @c nullptr otherwise.
+     */
+    ModelPtr library(const size_t &index);
+
+    /**
+     * @brief Manually add a local @c ModelPtr model instance to the importer library,
+     *        using the given @p key as a reference.
+     *
+     * If the given key already exists in the library, the function will return false
+     * and the library will not be changed.
+     * 
+     * @sa replaceModel
+     *
+     * @param model a @c ModelPtr instance to add.
+     * @param key a @c std::string representing the key to associate with the model.
+     *
+     * @return @c true if the model was added, @c false if it was not.
+     */
+    bool addModel(const ModelPtr &model, const std::string &key);
+
+    /**
+     * @brief Replace a @c ModelPtr model instance in the importer library,
+     *        using the given @p key as a reference.
+     *
+     * If the given key already exists in the library, the function will replace its
+     * model with the one supplied, and return @c true.
+     *
+     * If the given key does not exist, the function will return @c false, and
+     * the library will be unchanged.
+     *
+     * @sa addModel.
+     *
+     * @param model a @c ModelPtr instance to replace the current one.
+     * @param key a @c std::string representing the key at which to replace the model.
+     *
+     * @return @c true if the model was replaced, @c false if it was not.
+     */
+    bool replaceModel(const ModelPtr &model, const std::string &key);
+
+    /**
+     * @brief Retrieve the pair of URL key and import reference at the given index.
+     *
+     * This is taken from the list of dependencies for the models which have been resolved,
+     * and is what will break if those external files are ever moved or renamed.
+     *
+     * The first attribute of the returned pair is the URL at which the imported model was
+     * accessed and under which it is now stored in the library as its key, and the second attribute
+     * is the import reference.
+     *
+     * @return a @c std::pair of @c std::strings.
+     */
+    std::pair<std::string, std::string> externalDependency(size_t index) const;
+
+    /**
+     * @brief Get the number of external dependencies in the library.
+     *
+     * Returns the number of dependencies for the models which have been resolved by this
+     * importer.
+     *
+     * @return the number of external dependencies.
+     */
+    size_t externalDependencyCount() const;
+
+    /**
+     * @brief Clears the links with other models from all import sources.
```

Comment:
Requires non-brief documentation here as well.

Reply:
I thought we discussed this?  That there's no need to replicate brief documentation when it covers everything?

Reply:
I must have missed that discussion which  seems backwards.  If the documentation is only a single sentence then it shouldn't be labelled @brief.

Reply:
@brief implies there is more to come.

Reply:
So what's the preferred method when we have, say, three words to say?  Repeat it three times? Skip the @brief thing in favour of the main comment?  I guess I'll duplicate it all here, but it just seems like a waste of space :| Ah well.

Reply:
a0f083a9

Reply:
> That there's no need to replicate brief documentation when it covers everything

I think that discussion was that there is no need to repeat the description as a `@brief` if the brief would just be a copy of the first sentence of the full description (which is the doxygen default if no `@brief` is specified).